### ISSUE TITLE
Add special cases for UPPERCASE classes in GDScript syntax-coloring (`JSON`, `UPNP`, etc); and color `super` as a keyword

### ISSUE THREAD

**Issue by AlfishSoftware** [godotengine/godot-vscode-plugin#767]
### Godot version

4.3.stable

### VS Code version

1.96.0 codium snap

### Godot Tools VS Code extension version

2.3.0

### System information

Ubuntu 24.04

### Issue description

Opening an issue for the suggestion here https://github.com/godotengine/godot-vscode-plugin/pull/739#discussion_r1809465674 so it's not forgotten, as @DaelonSuzuka seems to agree.

It seems GDScript doesn't currently support semantic coloring, so it uses certain heuristics, like coloring UPPERCASE identifiers as constants. This is good enough for the majority of cases, but there's a few exceptions in Godot built-in classes that are somewhat easy to track.

These classes are known to be using UPPERCASE-only identifiers, exceptionally:
- `JSON` (constructible with new)
- `UPNP` (constructible with new)
- `OS`
- `IP`
- `JSONRPC`
- `XRVRS`

Since it's a very small list, it's easy to add them to `GDScript.tmLanguage.json` under `pascal_case_class` as exceptions, so they are matched as classes rather than constants. All 6 above should be added. If not, at least the first 2, which are constructible with new, and so it's extra weird to color them as constants.

https://github.com/godotengine/godot-vscode-plugin/blob/2e9117870dc0866925df947d963355ece7a32e14/syntaxes/GDScript.tmLanguage.json#L447-L450

---
Also, please move `super` away from `builtin_classes`, as it's a keyword, not a class.
It can be called like `super(arg1, arg2)` as well as `super.my_base_method(etc)`, and both of these semantics are different from a class, specially from a basic Variant type.
More importantly, it raises an error when used as an identifier, unlike built-in class names. So it's a keyword.

https://github.com/godotengine/godot-vscode-plugin/blob/2e9117870dc0866925df947d963355ece7a32e14/syntaxes/GDScript.tmLanguage.json#L439-L442

### Steps to reproduce

```gdscript
func wrong_coloring() -> void:
	var data_to_send: PackedStringArray = ["a", "b", "c"]
	var json_string: String = JSON.stringify(data_to_send)
	var json: JSON = JSON.new()
	var error: Error = json.parse(json_string)
	
	var upnp: UPNP = UPNP.new()
	upnp.discover()
	upnp.add_port_mapping(7777)
```
```gdscript
func expected_coloring() -> void: # For comparison
	var regex: RegEx = RegEx.new()
	regex.compile("\\w-(\\d+)")
	var regex2: RegEx = RegEx.create_from_string("\\w-(\\d+)")
```

**Comment by DaelonSuzuka** [comment]
Thanks for making the issue, that was a big help.

## UPPERCASE builtin classes

My instinct is to add these names to `builtin_classes`: 

https://github.com/godotengine/godot-vscode-plugin/blob/2e9117870dc0866925df947d963355ece7a32e14/syntaxes/GDScript.tmLanguage.json#L439-L442

This produces the desired highlighting, so is there a reason I should do it using `pascal_case_class` instead?

![Image](https://github.com/user-attachments/assets/e53647dd-daf8-4773-9439-0c4082e14a2d)

## super

Like this?

![Image](https://github.com/user-attachments/assets/86dad8a4-f30c-42da-908b-ea51e8d399c1)

**Comment by AlfishSoftware** [comment]
> This produces the desired highlighting, so is there a reason I should do it using `pascal_case_class` instead?

Yes! The theme ID is not exactly the same.
- `builtin_classes` is currently listing only variant/basic types (except for `super`)
- This makes it a bit similar to Godot editor, which does make a distinction between basic types, engine classes and user scripts. A theme could at least color variant types differently if it wants to.
- Except for RID, all UPPERCASE classes listed above are Object-derived. None are basic types. So conceptually, they're the same as other engine classes, they just so happen to be a "pascal-case" exception that has 1-letter words.

So, in summary, there's no reason to make a distinction between these and "proper" pascal-case classes (they're conceptually the same), but there is a valid reason, theme-wise, to separate them from basic types.

---

> super; Like this?

Yes. Though it might be best to be specific like `keyword.language.super.gdscript` in case a theme still wants to color it differently (e.g. if someone does prefer it colored like a type for whatever reason, e.g. got used to it). Regex group references like `$1` work in the name field too, so you could have it generic like `keyword.language.$1.gdscript` to add a specific id to many keywords.

**Comment by HolonProduction** [comment]
Those classes are handled in the grammar as `builtin_classes`. Should this be kept open to track whether this should have a different scope, or can we close it?

**Comment by AlfishSoftware** [comment]
I'd still argue it's conceptually incorrect to have them as `entity.name.type.class.builtin.gdscript` rather than `entity.name.type.class.gdscript` (which is what `pascal_case_class` uses). These theme ids are distinct and themes may want to distinguish them e.g. to color Variant types differently like Godot does.

Should probably be a simple change:
```json
		"pascal_case_class": {
			"match": "\\b(?:[A-Z]+(?:[a-z]+[A-Za-z0-9_]*)+|JSON|UPNP|OS|IP|JSONRPC|XRVRS)\\b",
			"name": "entity.name.type.class.gdscript"
		},
```
and move `const_vars` to after `pascal_case_class` in `base_expression`, so this one has priority.
Either that or make a new pattern with the same theme id
